File: new/common/src/main/java/net/shiroha233/roadweaver/persistence/sharded/RoadShardStorage.java
package net.shiroha233.roadweaver.persistence.sharded;

import net.minecraft.server.level.ServerLevel;
import net.shiroha233.roadweaver.core.model.RoadData;
import net.shiroha233.roadweaver.persistence.sqlite.RoadDatabaseManager;
import net.shiroha233.roadweaver.persistence.sqlite.RoadSqliteStorage;

import java.util.List;

/**
 * 道路数据存储门面，底层委托给 RoadSqliteStorage（H2 实现）
 */
public final class RoadShardStorage {
    private RoadShardStorage() {}

    public static void addRoad(ServerLevel level, RoadData rd) {
        RoadSqliteStorage.addRoad(level, rd);
    }

    public static List<RoadData> queryRect(ServerLevel level,
                                           int minBlockX, int minBlockZ,
                                           int maxBlockX, int maxBlockZ) {
        return RoadSqliteStorage.queryRect(level, minBlockX, minBlockZ, maxBlockX, maxBlockZ);
    }

    public static void flushAll(ServerLevel level) {
        RoadSqliteStorage.flushAll(level);
    }

    public static void clearAll(ServerLevel level) {
        RoadSqliteStorage.clearAll(level);
    }

    public static void shutdown() {
        RoadSqliteStorage.shutdown();
    }

    public static void closeConnection(ServerLevel level) {
        RoadDatabaseManager.checkpoint(level);
        RoadDatabaseManager.closeConnection(level);
    }
}

File: common/src/main/kotlin/net/shiroha233/roadweaver/persistence/sharded/RoadShardStorage.kt
package net.shiroha233.roadweaver.persistence.sharded

import net.minecraft.server.level.ServerLevel
import net.shiroha233.roadweaver.helpers.Records
import net.shiroha233.roadweaver.persistence.sqlite.RoadDatabaseManager
import net.shiroha233.roadweaver.persistence.sqlite.RoadSqliteStorage

/**
 * 道路数据存储（SQLite 实现）
 *
 * 此类现在是 RoadSqliteStorage 的门面（Facade），
 * 保持向后兼容的 API，同时底层使用 SQLite 数据库。
 */
object RoadShardStorage {
    @JvmStatic
    fun addRoad(level: ServerLevel, rd: Records.RoadData) {
        RoadSqliteStorage.addRoad(level, rd)
    }

    @JvmStatic
    fun queryRect(level: ServerLevel, minBlockX: Int, minBlockZ: Int, maxBlockX: Int, maxBlockZ: Int): List<Records.RoadData> {
        return RoadSqliteStorage.queryRect(level, minBlockX, minBlockZ, maxBlockX, maxBlockZ)
    }

    @JvmStatic
    fun flushAll(level: ServerLevel) {
        RoadSqliteStorage.flushAll(level)
    }

    @JvmStatic
    fun clearAll(level: ServerLevel) {
        RoadSqliteStorage.clearAll(level)
    }

    @JvmStatic
    fun shutdown() {
        RoadSqliteStorage.shutdown()
    }

    @JvmStatic
    fun closeConnection(level: ServerLevel) {
        RoadDatabaseManager.checkpoint(level)
        RoadDatabaseManager.closeConnection(level)
    }
}
